checker/src/main/java/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -162,6 +165,36 @@ private void addLowerBoundTypeFromValueType(
         }
     }
 
+    /** chars are unsigned implies chars are non-negative. See JLS 4.2. */
+    private void ensureCharNonNegative(AnnotatedTypeMirror type) {
+        TypeKind typeKind = type.getUnderlyingType().getKind();
+        switch (typeKind) {
+            case CHAR:
+                if (qualHierarchy.isSubtype(NN, type.getAnnotationInHierarchy(UNKNOWN)))
+                    type.replaceAnnotation(NN);
+                break;
+            case DECLARED:
+                String qualifiedName =
+                        TypesUtils.getQualifiedName((DeclaredType) type.getUnderlyingType())
+                                .toString();
+                if (qualifiedName.equals("java.lang.Character")) {
```

Comment:
This should be unnecessary now that `Character` is annotated as `@NonNegative` in the JDK